apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -73,7 +96,7 @@ public override bool Equals(object obj)
                     return false;
                 }
 
-                return obj is ClientPoolKey && Equals((ClientPoolKey) obj);
+                return obj is ClientCredentialKey other && Equals(other);
```

Comment:
hmm so you are saying we can get rid of the referenceequals.  I think thats true.